tests/src/ucs2.rs

```diff
@@ -0,0 +1,26 @@
+use uefi::{Status, Result};
+use uefi::ucs2::encode_ucs2;
+use alloc::Vec;
+
+pub fn ucs2_encoding_test() -> Result<()> {
+    let utf8_string = "őэ╋";
+    let mut ucs2_buffer : Vec<u16> = Vec::with_capacity(5);
```

Comment:
Why not use a local mutable array? Such as `let mut ucs2_buffer = [0u16; 5];`. We don't need to resize the vector.